src/main/kotlin/no/nav/familie/ef/iverksett/infrastruktur/advice/ApiExceptionHandler.kt

```diff
@@ -0,0 +1,36 @@
+package no.nav.familie.ef.iverksett.infrastruktur.advice
+
+import org.slf4j.LoggerFactory
+import org.springframework.core.NestedExceptionUtils
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+
+@Suppress("unused")
+@ControllerAdvice
+class ApiExceptionHandler {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    private fun rootCause(throwable: Throwable): String {
+        return NestedExceptionUtils.getMostSpecificCause(throwable).javaClass.simpleName
+    }
+
+    @ExceptionHandler(Throwable::class)
+    fun handleThrowable(throwable: Throwable): ResponseEntity<String> {
```

Comment:
Ønsker vi egentlig dette ? Hvis vi overstyrer f.eks en bad request og returnerer 500 i stedet ?

Reply:
For manuelt kastede feil så har vi ApiFeil nedenfor som kan gi statuskode videre.

I `ef-soknad-api` har vi løst det på denne måten;

```

@ExceptionHandler(Throwable::class)
    fun handleThrowable(throwable: Throwable): ResponseEntity<String> {
        val responseStatus = throwable::class.annotations.find { it is ResponseStatus }?.let { it as ResponseStatus }
        if (responseStatus != null) {
            return håndtertResponseStatusFeil(throwable, responseStatus)
        }
        return uventetFeil(throwable)
    }
    
    
    // Denne håndterer eks JwtTokenUnauthorizedException
    private fun håndtertResponseStatusFeil(throwable: Throwable,
                                           responseStatus: ResponseStatus): ResponseEntity<String> {
        val status = if (responseStatus.value != HttpStatus.INTERNAL_SERVER_ERROR) responseStatus.value else responseStatus.code
        val loggMelding = "En håndtert feil har oppstått" +
                          " throwable=${rootCause(throwable)}" +
                          " reason=${responseStatus.reason}" +
                          " status=$status"

        loggFeil(throwable, loggMelding)
        return ResponseEntity.status(status).body("Håndtert feil")
    }
    
    ```

Reply:
Prøvde å legge til dette, men hvis man fremprovoserer en bad request ved å sende ved feil i requesten f.eks, så ser det ikke ut som at ResponseStatus legges på i Throwable ? Stemmer det ?

Reply:
Satser på å bruke og videresende unntakstyper som sjekkes for i DefaultHandlerExceptionResolver, og sende 500 ellers